File: Aio-Algorithms-And-Solutions/Solutions/Exercism/CSharp/NeedForSpeed/RaceTrack.cs
namespace Aio_Algorithms_And_Solutions.Solutions.Exercism.NeedForSpeed;

class RaceTrack
{
    private readonly int _distance;
    
    public RaceTrack(int distance)
    {
        this._distance = distance;
    }

    public bool TryFinishTrack(RemoteControlCar car)
    {
        while (!car.BatteryDrained())
        {
            car.Drive();
            if (car.DistanceDriven() >= _distance) return true;
        }
        return false;
    }
}

File: Aio-Algorithms-And-Solutions/Solutions/Exercism/Java/NeedForSpeed/RaceTrack.java
package exercism;

class RaceTrack {
    private final int distance;

    public RaceTrack(int distance) {
        this.distance = distance;
    }

    public boolean tryFinishTrack(NeedForSpeed car) {
        while (car.distanceDriven() < distance) {
            if (car.batteryDrained()) return false;
            car.drive();
        }
        return true;
    }
}
